src/classes/CW_CampaignTriggerHandler.cls

```diff
@@ -0,0 +1,308 @@
+*
+*/
+
+//Cristina P: 25/06/14: Changing the class from "with sharing" to "without sharing" 
+//so the user can create related records while cost/brief is in approval
+public without sharing class CW_CampaignTriggerHandler 
+{
+	
+	private Boolean m_isExecuting = false;
+
+    public CW_CampaignTriggerHandler(Boolean isExecuting){
+        m_isExecuting = isExecuting;
+    }
+	
+	public void OnBeforeInsert(
+		List<CW_Campaign__c> newObjects, 
+		Map<Id,CW_Campaign__c> newObjectMap
+	)
+	{
+		stopNewCampaignCreationWhenRoadmapLocked(newObjects);
+	}
+	
+	public void OnAfterInsert( CW_Campaign__c[] newObjects, Map<Id,CW_Campaign__c> newObjectMap ) {
+		rollupPointsOnCampaignToRoadmap(newObjects);
+		createSingleSchedules(newObjects);
+    }
+    
+	public void OnBeforeUpdate( Map<Id,CW_Campaign__c> oldObjectMap, Map<Id,CW_Campaign__c> newObjectMap )
+	{
+		for(Id id : newObjectMap.keySet())
+		{
+			//stop user change points if roadmap is approved or in approval
+			if ((newObjectMap.get(id).Roadmap_Approval_Status__c==ConstantUtil.APPROVAL_PROCESS_APPROVED||
+            	newObjectMap.get(id).Roadmap_Approval_Status__c==ConstantUtil.APPROVAL_PROCESS_IN_APPROVAL) &&
+            	newObjectMap.get(Id).Points__c!=oldObjectMap.get(Id).Points__c)
+            	newObjectMap.get(id).addError(ConstantUtil.POINTS_CANNOT_BE_UPDATED);
+            	
+            if(newObjectMap.get(id).Roadmap__c != oldObjectMap.get(id).Roadmap__c)
+            {
+            	//stop user add a new campaign to a roadmap that is in approval or already approved
+            	stopNewCampaignCreationWhenRoadmapLocked(newObjectMap.Values()); 
+            	//stop user saving the campaign is roadmap is in approval or approved
+            	if (oldObjectMap.get(id).Roadmap_Approval_Status__c==ConstantUtil.APPROVAL_PROCESS_APPROVED||
+            	oldObjectMap.get(id).Roadmap_Approval_Status__c==ConstantUtil.APPROVAL_PROCESS_IN_APPROVAL)
+            		newObjectMap.get(id).addError(ConstantUtil.REMOVE_ROADMAP_ERROR);
+            }
+            //if brief is rejected or brief approval status has been changed to not submitted when brief was recalled 
+            //using standard recall functionality, change the campaign stage based on roadmap approval status	
+ 			if ((newObjectMap.get(id).Brief_Approval_Status__c==ConstantUtil.APPROVAL_PROCESS_REJECTED || 
+ 				newObjectMap.get(id).Brief_Approval_Status__c==ConstantUtil.APPROVAL_PROCESS_NOT_SUBMITTED) && 
+ 			    newObjectMap.get(id).Brief_Approval_Status__c!=oldObjectMap.get(id).Brief_Approval_Status__c)
+ 			{
+ 				if(oldObjectMap.get(id).Roadmap_Approval_Status__c == ConstantUtil.APPROVAL_PROCESS_APPROVED)
+ 					newObjectMap.get(id).Stage__c = ConstantUtil.CAMP_STAGE_ROADMAP_APPROVED;
+ 				else
+ 					newObjectMap.get(id).Stage__c = ConstantUtil.CAMP_STAGE_DRAFT;
+ 			}
+ 		}
+ 	}
+            	
+	public void OnAfterUpdate(
```

Comment:
Methods should use the `lowerCamelCase` naming convention. I suggest to rename accordingly.